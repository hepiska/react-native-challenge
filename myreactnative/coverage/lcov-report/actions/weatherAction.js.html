<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for actions/weatherAction.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">actions</a> weatherAction.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.86% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/4</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import axios from 'axios';
import { FETCH_WEATHER } from './constans';
&nbsp;
export const fetchWeatherDone = weathers =&gt; ({ type: FETCH_WEATHER, payload: weathers });
&nbsp;
export const fetchWeather = <span class="fstat-no" title="function not covered" >ci</span>ty =&gt; <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >di</span>spatch =&gt; (<span class="cstat-no" title="statement not covered" >axios.get('https://api.apixu.com/v1/forecast.json?key=146af849a55b4551877173858170205&amp;q=' + city + '&amp;days=2', {})</span></span>
.then(<span class="fstat-no" title="function not covered" >re</span>sponse =&gt; (<span class="cstat-no" title="statement not covered" >dispatch(fetchWeatherDone(response.data)))</span>)
.catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; <span class="cstat-no" title="statement not covered" >console.log(err))</span>
);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri May 05 2017 01:33:02 GMT+0700 (WIB)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
