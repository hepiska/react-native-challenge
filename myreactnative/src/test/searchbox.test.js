import React from 'react';
import { shallow, render, mount } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../store'
import ReactDOM from 'react-dom';
import sinon from 'sinon';
import {
  Text,
  View,
} from 'react-native';
import SearchBox, { SearchBox as SearchBox1 } from '../components/homeComponent/searchBox';
import { SearchBar, Button } from 'react-native-elements';
import { connect } from 'react-redux';
import { fetchWeather } from '../actions';


describe('searchBox component', () => {
  const searchcomp = shallow(<SearchBox1 />);
  //const searchBoxMount = mount(<Provider store={store}><SearchBox1 /></Provider>)
  it('rendered', () => {
    expect(searchcomp).toMatchSnapshot();
  });
  it('search on change', () => {
    searchcomp.find(SearchBar).simulate('ChangeText', 'jakarta' );
    expect(searchcomp.state('text')).toEqual('jakarta');
  });
  it('search button click', () => {
    const press = sinon.spy();
    // searchBoxMount.find(Button).toHaveLength(1);
    //console.log('---------- render',searchcomp.find(Button).simulate('Press'))
    // console.log('---------- render',searchBoxMount.find(Button));
    //expect(searchcomp.state('text')).toEqual('');
  })
})
