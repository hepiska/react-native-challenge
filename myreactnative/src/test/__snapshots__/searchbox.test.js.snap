// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchBox component rendered 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "height": "16%",
            "width": "100%",
          }
    }
>
    <Search
        icon={Object {}}
        inputStyle={undefined}
        lightTheme={true}
        loadingIcon={Object {}}
        noIcon={false}
        onChangeText={[Function]}
        placeholder="Type Here..."
        placeholderTextColor="#86939e"
        round={false}
        showLoadingIcon={false}
        value=""
    />
    <Button
        buttonStyle={
            Object {
                "marginLeft": 0,
                "marginRight": 0,
              }
        }
        icon={
            Object {
                "name": "search",
              }
        }
        onPress={[Function]}
        raised={true}
        small={true}
        title="Search"
    />
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "height": "16%",
                  "width": "100%",
                }
      }
>
      <Search
            icon={Object {}}
            inputStyle={undefined}
            lightTheme={true}
            loadingIcon={Object {}}
            noIcon={false}
            onChangeText={[Function]}
            placeholder="Type Here..."
            placeholderTextColor="#86939e"
            round={false}
            showLoadingIcon={false}
            value=""
      />
      <Button
            buttonStyle={
                  Object {
                        "marginLeft": 0,
                        "marginRight": 0,
                      }
            }
            icon={
                  Object {
                        "name": "search",
                      }
            }
            onPress={[Function]}
            raised={true}
            small={true}
            title="Search"
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SearchBox />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SearchBox {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "text": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "height": "16%",
                              "width": "100%",
                            }
          }
>
          <Search
                    icon={Object {}}
                    inputStyle={undefined}
                    lightTheme={true}
                    loadingIcon={Object {}}
                    noIcon={false}
                    onChangeText={[Function]}
                    placeholder="Type Here..."
                    placeholderTextColor="#86939e"
                    round={false}
                    showLoadingIcon={false}
                    value=""
          />
          <Button
                    buttonStyle={
                              Object {
                                        "marginLeft": 0,
                                        "marginRight": 0,
                                      }
                    }
                    icon={
                              Object {
                                        "name": "search",
                                      }
                    }
                    onPress={[Function]}
                    raised={true}
                    small={true}
                    title="Search"
          />
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Object {
                              "height": "16%",
                              "width": "100%",
                            }
          }
>
          <Search
                    icon={Object {}}
                    inputStyle={undefined}
                    lightTheme={true}
                    loadingIcon={Object {}}
                    noIcon={false}
                    onChangeText={[Function]}
                    placeholder="Type Here..."
                    placeholderTextColor="#86939e"
                    round={false}
                    showLoadingIcon={false}
                    value=""
          />
          <Button
                    buttonStyle={
                              Object {
                                        "marginLeft": 0,
                                        "marginRight": 0,
                                      }
                    }
                    icon={
                              Object {
                                        "name": "search",
                                      }
                    }
                    onPress={[Function]}
                    raised={true}
                    small={true}
                    title="Search"
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SearchBox />,
}
`;
